pkg/controller/provider/validation.go

```diff
@@ -173,6 +173,30 @@ func (r *Reconciler) validateURL(provider *api.Provider) error {
 	return nil
 }
 
+func validateConnectionStatus(r *Reconciler, provider *api.Provider, secret *core.Secret) {
+	if base.GetInsecureSkipVerifyFlag(secret) {
+		provider.Status.SetCondition(libcnd.Condition{
+			Type:     ConnectionInsecure,
+			Status:   True,
+			Reason:   SkipTLSVerification,
+			Category: Warn,
+			Message:  "TLS is susceptible to machine-in-the-middle attacks when certificate verification is skipped.",
+		})
+	} else {
+		if crt, err := r.Reconciler.VerifyTLSConnection(provider.Spec.URL, secret); err == nil {
+			provider.Status.Fingerprint = util.Fingerprint(crt)
```

Comment:
This applies only to the vmware please add check if it's vmware provider as this is general method for all providers.